### PR TITLE
[WIP] handle different versions for configuration modules

### PR DESCRIPTION
**Pull request by MioRtia**
Handling different versions for configuration modules
* every config modules implement Versionable
* the default version is 1.1
* if the newest handled version of a module is 1.1, any newer version will be handled as 1.1

### CODE REVIEW

**Comment by mathbagu** [comment]
Add a local variable `version` in all the modules and initialize this variable in the `load` method.
For modules which we already manage two different versions of properties or module names, initialize the version to 1.0 or 1.1

File: afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/SecurityConfig.java [hunk 1]
```diff
@@ -62,4 +78,14 @@ public boolean isSkipTokenValidityCheck() {
...
     }
+
+    @Override
+    public String getName() {
+        return CONFIG_MODULE_NAME;
+    }
+
+    @Override
+    public String getVersion() {
+        return this.version.toString();
```
Comment (yichen88) [inline afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/SecurityConfig.java]: "this" is not necessary here

File: commons/src/main/java/com/powsybl/commons/config/VersionConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+public class VersionConfig {
+
+    private final String[] version;
+
+    public VersionConfig(String version) {
+        this.version = version.split("\\.");
+    }
+
+    public boolean isStrictlyOlderThan(String version) {
+        String[] otherVersion = version.split("\\.");
```
Comment (yichen88) [inline commons/src/main/java/com/powsybl/commons/config/VersionConfig.java]: Could you add some junit tests for these two methods?

File: commons/src/main/java/com/powsybl/commons/config/VersionConfig.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.config;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public class VersionConfig {
+
```
Comment (yichen88) [inline commons/src/main/java/com/powsybl/commons/config/VersionConfig.java]: i don't think it is a good name. Maybe ConfigVersion or VersionNumber? (Note: Version is already used as Interface in powsybl)

**Pull request closed without merging**